lib/private/Files/FilenameValidator.php

```diff
@@ -0,0 +1,192 @@
+<?php
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OC\Files;
+
+use OCP\Files\IFilenameValidator;
+use OCP\IConfig;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @since 30.0.0
+ */
+class FilenameValidator implements IFilenameValidator {
+
+	/**
+	 * @var list<string>
+	 */
+	private array $forbiddenNames;
```

Comment:
```suggestion
	private array $forbiddenNames = [];
```
Either init to empty, or change for nullable and init with `null` and check for `null` instead of `empty()`.